contrib/meson/meson/tests/meson.build

```diff
@@ -10,43 +10,78 @@
 
 lz4_source_root = '../../../..'
 
+fuzzer_time = 90
 exes = {
-  'fullbench': {
-    'sources': files(lz4_source_root / 'tests/fullbench.c'),
+  'abiTest': {
+    'sources': files(lz4_source_root / 'tests/abiTest.c'),
+    'test': false,
+  },
+  'checkFrame': {
+    'sources': files(lz4_source_root / 'tests/checkFrame.c'),
     'include_directories': include_directories(lz4_source_root / 'programs'),
   },
-  'fuzzer': {
-    'sources': files(lz4_source_root / 'tests/fuzzer.c'),
+  'checkTag': {
+    'sources': files(lz4_source_root / 'tests/checkTag.c'),
+    'test': false,
+  },
+  'datagen': {
+    'sources': files(lz4_source_root / 'tests/datagencli.c'),
+    'objects': lz4.extract_objects(lz4_source_root / 'programs/datagen.c'),
     'include_directories': include_directories(lz4_source_root / 'programs'),
   },
+  'decompress-partial-usingDict.c': {
+    'sources': files(lz4_source_root / 'tests/decompress-partial-usingDict.c'),
+  },
+  'decompress-partial.c': {
+    'sources': files(lz4_source_root / 'tests/decompress-partial.c'),
+  },
   'frametest': {
     'sources': files(lz4_source_root / 'tests/frametest.c'),
     'include_directories': include_directories(lz4_source_root / 'programs'),
+    'args': ['-v', '-T@0@s'.format(fuzzer_time)],
+    'test': false,
   },
-  'roundTripTest': {
-    'sources': files(lz4_source_root / 'tests/roundTripTest.c'),
-  },
-  'datagen': {
-    'sources': files(lz4_source_root / 'tests/datagencli.c'),
-    'objects': lz4.extract_objects(lz4_source_root / 'programs/datagen.c'),
+  # 'freestanding': {
```

Comment:
I disabled this because I cannot get Meson to recognize `-ffreestanding -nostdlib`. @eli-schwartz I am tempted to open a bug with Meson.

Reply:
As I pointed out already in our other discussion, simply don't use cc.get_supported_arguments, and just check the compiler id. Verifying -nostdlib requires a custom check, because ld itself errors out when you try to compile test code that isn't itself freestanding...

Reply:
```meson
if [compiler_gcc, compiler_clang].contains(cc_id)
    exes += {
        'freestanding': {.....}
    }
endif
```